### PR TITLE
Prevent polymorphic slots causing content evaluation

### PR DESCRIPTION
**Pull request by BlakeWilliams**
Currently there's a bug in polymorphic slots where setting a slot can
cause `content` to be evaluated early. This requires code to be order
dependent, setting content and other values before setting a polymorphic
slot. This primarily occurs in code that makes heavy use of component
API's within Ruby, not within typical ERB usage.

This updates the polymorphic slot code to avoid the `get_slot` method so
that content won't be evaluated early.

### CODE REVIEW

**Review by camertron — APPROVED** [review]
Awesome, thank you! 🚀

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -18,8 +22,6 @@ nav_order: 5
 
     *Cameron Dutro*
 
-## 2.75.0
-
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:22]: Did you mean to remove these lines?

Comment (BlakeWilliams) [reply docs/CHANGELOG.md:22]: Ah nope, bad rebase. Fixing!

**Review by joelhawksley — APPROVED** [review]

**Review by joelhawksley — APPROVED** [review]

**Comment by BlakeWilliams** [comment]
I have no idea why the linter started failing on this PR. Maybe related to https://github.com/ViewComponent/view_component/pull/1576?

**Pull request merged**